plugins/source-github/test/e2e/basic_workflow_test.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2019 The Knative Authors
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build e2e
+// +build !eventing
+
+package e2e
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"gotest.tools/assert"
+
+	"knative.dev/client/lib/test"
+	"knative.dev/client/pkg/util"
+
+	commone2e "github.com/maximilien/kn-source-pkg/test/e2e"
+)
+
+type e2eTest struct {
+	it *commone2e.E2ETest
+}
+
+func newE2ETest(t *testing.T) *e2eTest {
+	currentDir, err := os.Getwd()
+	assert.NilError(t, err)
+
+	it, err := commone2e.NewE2ETest("kn-source_github", filepath.Join(currentDir, "../.."), false)
+	assert.NilError(t, err)
+	defer func() {
+		assert.NilError(t, it.KnTest().Teardown())
+	}()
+
+	e2eTest := &e2eTest{
+		it: it,
+	}
+
+	return e2eTest
+}
+
+func TestBasicWorkflow(t *testing.T) {
+	t.Parallel()
+
+	e2eTest := newE2ETest(t)
+	assert.Assert(t, e2eTest != nil)
+
+	r := test.NewKnRunResultCollector(t, e2eTest.it.KnTest())
+	defer r.DumpIfFailed()
+
+	err := e2eTest.it.KnPlugin().Install()
+	assert.NilError(t, err)
+
+	t.Log("kn-source_github create 'source-name' with 'sink-name'")
+	e2eTest.ghSourceCreate(t, r, "source-name", "sink-name")
```

Comment:
I wonder where did you create the sink `sink-name`. I run test locally, and it reported:
```
$ kn source_github create source-name --sink sink-name --namespace kne2etests0 --namespace kne2etests0
f.KnSourceParams(): &types.KnSourceParams{KnParams:commands.KnParams{Output:io.Writer(nil), KubeCfgPath:"", ClientConfig:(*clientcmd.DeferredLoadingClientConfig)(0xc0000ec640), NewServingClient:(func(string) (v1.KnServingClient, error))(0x1eb7e20), NewSourcesClient:(func(string) (v1alpha2.KnSourcesClient, error))(0x1eb7ea0), NewEventingClient:(func(string) (v1alpha1.KnEventingClient, error))(0x1eb7f20), NewDynamicClient:(func(string) (dynamic.KnDynamicClient, error))(0x1eb7fa0), LogHTTP:false, fixedCurrentNamespace:""}, SinkFlag:flags.SinkFlags{sink:"sink-name"}}
panic: runtime error: invalid memory address or nil pointer dereference
```

I think it is because that `sink-name` is not created.